### PR TITLE
feat(neptune): Additional Neptune checks

### PR DESCRIPTION
**Pull request by sansns**
### Context

1. Check for backups being enabled
2. Deletion protection being enabled
3. IAM authentication being enabled
4. Multi AZ is enabled
5. Storage is encrypted at rest

### Description

AWS Neptune Controls: https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html
Trend Conformity: https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/Neptune/

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by snaow** [comment]
Thanks @sansns! We will review and merge/get back to you soon.

File: prowler/providers/aws/services/neptune/neptune_cluster_backup_enabled/neptune_cluster_backup_enabled.metadata.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "aws",
+  "CheckID": "neptune_cluster_backup_retention",
+  "CheckTitle": "Check if Neptune Clusters have backup enabled.",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_cluster_backup_enabled/neptune_cluster_backup_enabled.metadata.json]:
```suggestion
  "CheckTitle": "Check for Neptune Clusters Backup Retention Period.",
```

File: prowler/providers/aws/services/neptune/neptune_cluster_backup_enabled/neptune_cluster_backup_enabled.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            report.resource_id = cluster.name
+            report.resource_arn = cluster.arn
+            report.resource_tags = cluster.tags
+            report.status = "FAIL"
+            report.status_extended = (
+                f"Neptune Cluster {cluster.name} does not have backup enabled."
+            )
+            if (
+                cluster.backup_retention_period > 0
+                and cluster.backup_retention_period < 7
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_cluster_backup_enabled/neptune_cluster_backup_enabled.py]: Could you do this as a configurable variable, please?

File: prowler/providers/aws/services/neptune/neptune_service.py [hunk 3]
```diff
@@ -39,6 +39,12 @@ def __describe_clusters__(self, regional_client):
                         arn=cluster_arn,
                         name=cluster["DBClusterIdentifier"],
                         id=cluster["DbClusterResourceId"],
+                        backup_retention_period=cluster["BackupRetentionPeriod"],
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_service.py]:
```suggestion
                        backup_retention_period=cluster.get("BackupRetentionPeriod",0),
```

File: prowler/providers/aws/services/neptune/neptune_service.py [hunk 4]
```diff
@@ -96,6 +102,14 @@ class Cluster(BaseModel):
     arn: str
     name: str
     id: str
+    backup_retention_period: int = 0
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_service.py]:
```suggestion
    backup_retention_period: int
```

File: prowler/providers/aws/services/neptune/neptune_service.py [hunk 5]
```diff
@@ -96,6 +102,14 @@ class Cluster(BaseModel):
...
+    encrypted: Optional[bool]
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_service.py]:
```suggestion
    encrypted: bool
```

File: prowler/providers/aws/services/neptune/neptune_service.py [hunk 6]
```diff
@@ -96,6 +102,14 @@ class Cluster(BaseModel):
...
+    kms_key: Optional[str]
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_service.py]:
```suggestion
    kms_key: str
```

File: prowler/providers/aws/services/neptune/neptune_service.py [hunk 7]
```diff
@@ -96,6 +102,14 @@ class Cluster(BaseModel):
...
+    multi_az: bool
+    iam_auth: bool
+    deletion_protection: bool
+    cluster_id: Optional[str]
+    cluster_arn: Optional[str]
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_service.py]: What are these for?

File: prowler/providers/aws/services/neptune/neptune_cluster_deletion_protection/neptune_cluster_deletion_protection.metadata.json [hunk 8]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "aws",
+  "CheckID": "neptune_cluster_deletion_protection_enabled",
+  "CheckTitle": "Check if Neptune Clusters storage has deletion protection enabled.",
+  "CheckType": [],
+  "ServiceName": "neptune",
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:aws:rds:region:account-id:db-cluster",
+  "Severity": "medium",
+  "ResourceType": "AwsRdsDbClusters",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_cluster_deletion_protection/neptune_cluster_deletion_protection.metadata.json]:
```suggestion
  "ResourceType": "AwsRdsDbCluster",
```

File: prowler/providers/aws/services/neptune/neptune_cluster_iam_authentication_enabled/neptune_cluster_iam_authentication_enabled.metadata.json [hunk 9]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "aws",
+  "CheckID": "neptune-cluster-iam-database-authentication",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_cluster_iam_authentication_enabled/neptune_cluster_iam_authentication_enabled.metadata.json]:
```suggestion
  "CheckID": "neptune_cluster_iam_authentication_enabled",
```

File: prowler/providers/aws/services/neptune/neptune_cluster_multi_az/neptune_cluster_multi_az.metadata.json [hunk 10]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "aws",
+  "CheckID": "neptune-cluster-multi-az-enabled",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_cluster_multi_az/neptune_cluster_multi_az.metadata.json]:
```suggestion
  "CheckID": "neptune_cluster_multi_az_enabled",
```

File: prowler/providers/aws/services/neptune/neptune_cluster_storage_encrypted/neptune_cluster_storage_encrypted.metadata.json [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+  "SubServiceName": "",
+  "ResourceIdTemplate": "arn:aws:rds:region:account-id:db-cluster",
+  "Severity": "high",
+  "ResourceType": "AwsRdsDbClusters",
+  "Description": "Check if Neptune Clusters storage is encrypted at rest.",
+  "Risk": "Ensure that the data available on your Amazon Neptune database instances is encrypted in order to meet regulatory requirements and prevent unauthorized users from accessing sensitive information. Encryption provides an additional layer of protection by securing your Neptune databases from unauthorized access to the underlying storage. Neptune is a fast, scalable, highly secure and fully-managed graph database service that makes it easy to build and run applications that work with deeply connected datasets.",
+  "RelatedUrl": "https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-1",
+  "Remediation": {
+    "Code": {
+      "CLI": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/Neptune/encryption-enabled.html",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_cluster_storage_encrypted/neptune_cluster_storage_encrypted.metadata.json]:
```suggestion
      "CLI": "https://docs.prowler.com/checks/aws/general-policies/general_18/",
```

File: prowler/providers/aws/services/neptune/neptune_cluster_storage_encrypted/neptune_cluster_storage_encrypted.metadata.json [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+      "NativeIaC": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/Neptune/encryption-enabled.html",
+      "Other": "",
+      "Terraform": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/Neptune/encryption-enabled.html"
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/neptune/neptune_cluster_storage_encrypted/neptune_cluster_storage_encrypted.metadata.json]:
```suggestion
      "Terraform": "https://docs.prowler.com/checks/aws/general-policies/general_18/"
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Love this @sansns ! Please, review my comments.

File: prowler/providers/aws/services/neptune/neptune_cluster_backup_enabled/neptune_cluster_backup_enabled.py (hunk 2 above)
Comment (sansns) [reply prowler/providers/aws/services/neptune/neptune_cluster_backup_enabled/neptune_cluster_backup_enabled.py]: Do you have an example to reference? I want to make sure Im following the code style so it does not cause any compatibility issues. Will also need to add some additional tests cases for this as well.

**Comment by sansns** [comment]
One outstanding review left, but other ones have been addressed.

File: prowler/providers/aws/services/neptune/neptune_cluster_backup_enabled/neptune_cluster_backup_enabled.py (hunk 2 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/neptune/neptune_cluster_backup_enabled/neptune_cluster_backup_enabled.py]: Sure! You can find it here: https://github.com/prowler-cloud/prowler/blob/9ca64e7bdb46d4f5791140a37946ca8ae98d4ccf/prowler/providers/aws/services/acm/acm_certificates_expiration_check/acm_certificates_expiration_check.py#L12

Comment (sansns) [reply prowler/providers/aws/services/neptune/neptune_cluster_backup_enabled/neptune_cluster_backup_enabled.py]: Thank you for the example. Added that into the check and an additional test case as well to verify it was being successfully set.

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**